framework/archiver/exceptions.py

```diff
@@ -0,0 +1,69 @@
+from framework.archiver import mails
+from framework.archiver.utils import delete_registration_tree
+
+from website import settings
+from website.mails import send_mail
+
+def send_archiver_mail(*args, **kwargs):
+    """A proxy to facilitate unit testing"""
+    send_mail(*args, **kwargs)
+
+class ArchiverError(Exception):
+    pass
+
+class ArchiverSizeExceeded(ArchiverError):
+    def __init__(self, src, dst, user, stat_result, *args, **kwargs):
+        """Capture and respond to an archive task that is failed due to disk usage constraints.
+        Deletes the failed registration.
+
+        :param src: Node being registered
+        :param dst: registration Node
+        :param user: registration initiator
+        :param stat_result: AggregateStatResult of Node addon metadata
+        """
+        super(ArchiverSizeExceeded, self).__init__(*args, **kwargs)
+        send_archiver_mail(
```

Comment:
This should be caught somewhere and then send the emails.
Side effects from class instantiation should be avoided.